core/src/main/java/org/mule/runtime/core/api/policy/PolicyChain.java

```diff
@@ -115,20 +117,20 @@ public CoreEvent process(CoreEvent event) throws MuleException {
 
   @Override
   public Publisher<CoreEvent> apply(Publisher<CoreEvent> publisher) {
-    return from(publisher)
+    return Flux.from(publisher)
+        .doOnNext(pushBeforeNextFlowStackElement()
+            .andThen(req -> ((BaseEventContext) req.getContext())
+                .onResponse((resp, t) -> popFlowFlowStackElement().accept(req)))
+            .andThen(notificationHelper.notification(PROCESS_START)))
+        // TODO MULE-16370 remove this flatMap
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7629&issues=AWiGlnVfotIXnmsUoniQ&open=AWiGlnVfotIXnmsUoniQ)

[ISSUE_KEY]: AWiGlnVfotIXnmsUoniQ